kubetest2/internal/deployers/eksapi/deployer.go

```diff
@@ -260,6 +260,9 @@ func (d *deployer) verifyUpFlags() error {
 	if d.UnmanagedNodes && d.EFA && len(d.InstanceTypes) != 1 {
 		return fmt.Errorf("--efa requires a single instance type")
 	}
+	if d.AMIType != "" && d.AMI != "" {
+		return fmt.Errorf("Cannot accept mutually exclusive flags --ami-type and --ami because it implies conflicting deployment modes")
```

Comment:
need some suggestion on this message/condition. we don't validate you *can't* provide the `--ami` flag if you're not using unmanaged nodes, so not sure whether enforcing this is useful in practice @cartermckinnon

Reply:
validating each on their own according to `--unmanaged-nodes` fixes the mutual exclusion and i think is clearer. small update with this